### DISCUSSION TITLE
How to disable directory listing in Next js

### DISCUSSION THREAD

**Discussion by Ebdulmomen1** [vercel/next.js#14637] [Help]
I have deployed website built in `Next.js` and apparently when you deploy a next js app you can find the directory of all the static files and resources which i don't feel very good about.

the directory is generated in this way:  `mydomain.com/_next/static` in my case `https://www.ovanya.com/_next/static/`

the next js <a href="https://vercel.com/docs/directory-listing">docs</a> say: 

> To prevent showing a Directory Listing page, add an index file, such as index.html, which will then show that file instead when accessing the directory's path.

but i really don't understand what directory should i add the index.html file to? so where should i exatcly put the index.html at?

and another question: is it bad if i leave this just like that? does it have any risks or stuff like that?

**Comment by IamLizu** [comment]
> i really don't understand what directory should i add the index.html file to

You should add an empty `index.html` in the directory where you want to turn off directory listing. 

For example, let's say i have a domain `example.com` and a directory named `my-files` in it. And I want to turn off directory list of `my-files` dir, so that no one can see what files or sub-directories are there from the browser. So I will need to add a blank `index.html` file inside of this `my-files` directory.



> is it bad if i leave this just like that? does it have any risks or stuff like that?

For a production site, its better to turn of directory listing. Imagine you have important files in a directory and its open to everyone!

**Comment by kimroen** [comment]
Hi, sorry to bump this older thread, but we have the same issue. Adding index-files doesn't seem to help much, because they are fingerprinted, so the built folder inside `_next/static` doesn't actually have an `index.html` file, and you're able to see a directory listing.

**Reply by Rahul-Sharma-87** [reply]
yes we are also facing the same issue this is a security bug..

**Comment by anandhuremanan** [comment]
Did anyone find a solution for this?

**Comment by ahsalar** [comment]
I came across this discussion while googling to find a solution for another issue. Vercel is now disabling directory listing by default. You can toggle the default behavior in the [advanced settings section](https://vercel.com/docs/edge-network/directory-listing).
